lib/convert_list_to_deb822.py

```diff
@@ -0,0 +1,67 @@
+#!/usr/bin/env python3
+
+"""
+This script is called after running do-release-upgrade in a machine.
+This converts list files to deb822 files when upgrading to Noble.
+"""
+
+import logging
+import os
+import sys
+
+from aptsources.sourceslist import SourceEntry  # type: ignore
+
+from uaclient import entitlements
+from uaclient.apt import _get_sources_file_content
+from uaclient.cli import setup_logging
+from uaclient.config import UAConfig
+from uaclient.system import (
+    ensure_file_absent,
+    get_release_info,
+    load_file,
+    write_file,
+)
+from uaclient.util import set_filename_extension
+
+if __name__ == "__main__":
+    series = get_release_info().series
+    if series != "noble":
+        sys.exit(0)
+
+    setup_logging(logging.DEBUG)
+    cfg = UAConfig()
+
+    for entitlement_class in entitlements.ENTITLEMENT_CLASSES:
+        if not issubclass(
+            entitlement_class, entitlements.repo.RepoEntitlement
+        ):
+            continue
+
+        entitlement = entitlement_class(cfg)
+
+        filename = set_filename_extension(entitlement.repo_file, "list")
+        if os.path.exists(filename):
```

Comment:
@renanrodrigo let's just add a comment that `do-release-upgrade` will comment out the list file of other services (i.e. `usg`) and because of that, they not be evaluated here

Reply:
Added